sources/lib/Model/ModelTrait/WriteQueries.php

```diff
@@ -96,6 +96,24 @@ public function updateByPk(array $primaryKey, array $updates): ?FlexibleEntityIn
         $where = $this
             ->checkPrimaryKey($primaryKey)
             ->getWhereFrom($primaryKey);
+
+        $iterator = $this->updateWhere($where, $updates);
+
+        if ($iterator->isEmpty()) {
+            return null;
+        }
+
+        return $iterator->current();
```

Comment:
on peut pas faire ça directement ?
```suggestion
        return $this->updateWhere($where, $updates)->current();
```